modules/internal/BytesStringCommon.chpl

```diff
@@ -183,6 +184,66 @@ module BytesStringCommon {
     return ret;
   }
 
+   /*
+   This function decodeHelp is used to create a wrapper for 
+   qio_decode_char_buf* and qio_decode_char_buf_esc and return 
+   the value of syserr , cp and nbytes.
+   asks for buffer , size of buffer , starting index to read buffer,
+   and choice between "qio_decode_char_buf" 
+   and "qio_decode_char_buf_esc" for escape statement
+   */
+
+   proc decodeHelp(buff:c_ptr(uint(8)), buflen:int, 
+                  bufstart:int, allowEsc: bool )
+   throws {
+    
+    //pragma "fn synchronization free"
+    extern proc qio_decode_char_buf(ref chr:int(32), 
+                                    ref nbytes:c_int,
+                                    buf:c_string,
+                                    buflen:ssize_t): syserr;
+    //pragma "fn synchronization free"
+    extern proc qio_decode_char_buf_esc(ref chr:int(32),
+                                        ref nbytes:c_int,
+                                        buf:c_string,
+                                        buflen:ssize_t): syserr;
+    // esc chooses between qio_decode_char_buf_esc and
+    // qio_decode_char_buf as a single wrapper function 
+    if(allowEsc){
+      var chr: int(32);
+      var nbytes: c_int;
+      var start = bufstart:c_int;
+      //buff + start + 1 is there since difference between 0 and 1 based indexing
+      var multibytes = (buff + start+1): c_string;
```

Comment:
Observation, make check fails if I have somthing like 
`var mulitbytes = (buff + start) : c_string;
(Probably I think has to do with difference in indexing )

Complete error message
```
[Info] Running minimal test script: $CHPL_HOME/util/test/checkChplInstall
[Info] Found executable chpl in /usr/local/bin/chpl.
[Info] Found $CHPL_HOME directory: /home/parthsarthiprasad/Desktop/github/chapel
[Info] /home/parthsarthiprasad/.chpl does not exist. Creating it.
[Info] Temporary test job directory: /home/parthsarthiprasad/.chpl/chapel-test-sMUrE
[Info] Compiling $CHPL_HOME/test/release/examples/hello6-taskpar-dist.chpl
[Info] Compiling with C backend
[Info] Test job compiled into /home/parthsarthiprasad/.chpl/chapel-test-sMUrE/hello6-taskpar-dist
[Info] $CHPL_LAUNCHER=none is compatible with test script.
[Info] Running test job.
[Info] Test job complete.
[Fail] There was an issue with the installation, test job output incorrect.
[Info] To see the test outputs, export CHPL_CHECK_DEBUG=1 and re-run "make check"

== Actual Test Output (raw, with verbose) ==
QTHREADS: Using 4 Shepherds
QTHREADS: Using 1 Workers per Shepherd
QTHREADS: Guard Pages Enabled
QTHREADS: Using 8376320 byte stack size.
executing on a single node
<command line setting of 'printLocaleName'>: error: halt reached - range 0..5 out of bounds for string with 5 bytes
make: *** [Makefile:204: check] Error 20
```

Reply:
I don't think that +1 should be there. I am wondering if this was because some issue due to recent changes in 0-based indexing. Is this branch rebased on current master? If so, did you rebuild the compiler and the runtime (a simple `make -j8`, really)?

Reply:
Yep the branch has been up to date at this point, I did rebuilt the compiler before `make check` before pushing to the branch i.e 10 days back . Rebuilt with `make` and make `install` ,
 without that `+1` It falls into make check error.

Reply:
I'll checkout this branch and take a look myself to understand the issue better. Will keep you posted.